src/app.js

```diff
@@ -6,8 +6,294 @@ import _ from 'underscore';
+        error: failedSave,
+      });
+  } else {
+      // getting here means there were client-side validation errors reported
+      updateStatusMessageFrom(trip.validationError);
+  }
+
+  setTimeout(function(){ $('#status-messages').hide(); }, 10000);
+
+};
+
+//ADDING A Reservation
+const addReservationHandler = function(event) {
+  event.preventDefault();
+  const reservationData = {};
+  ['name', 'email', 'age'].forEach((field) => {
+    // select the input corresponding to the field we want
+    const inputElement = $(`#reservation-form input[name="${ field }"]`);
+    const value = inputElement.val();
+    reservationData[field] = value;
+
+    inputElement.val('');
+  });
+
+  console.log("Read Reservation data");
+  console.log(reservationData);
+
+  reservationData["trip_id"] = $(this).data("id");
+  const reservation = new Reservation(reservationData);
+
+  console.log(reservation.url);
+
+  // reservation.save({}, {
+  if (reservation.save) {
+    $('#status-messages ul').empty();
+    $('#status-messages ul').append($(`<li>${reservation.get('name')} successfully reserved for this trip!</li>`)).html();
+  } else {
+    $('#status-messages ul').empty();
+    $('#status-messages ul').append($(`<li>Unable to complete reservation for ${reservation.get('name')}. Please try again.</li>`)).html();
+  }
+
+    // success: (model, response) => {
+    //   console.log('Successfully saved reservation!');
+    //   // $('#status-messages').html('Successfully saved reservation!');
+    //   // updateStatusMessageWith(`${model.get('name')} has been reserved for this trip.`);
+    //   successfullSave;
+    // },
+    // error: (model, response) => {
+    //   console.log('Failed to save reservation! Server response:');
+    //   console.log(response);
+    //   // $('#status-messages').html('Failed to save reservation!');
+    //   failedSave;
+      // FIGURE OUT HOW TO ADD EACH ERROR
+      //SET TIMEOUT ON THIS ONE AND THE SUCCESS - check TREK Project
+
+
+};
+
+
+$(document).ready(() => {
+
+  // $('#reservation-form').hide()
+  $('#all-trips').hide();
+  $('#add-trip').hide();
+
+  // ERROR AND SUCCESS MESSAGES
+  statusMessagesTemplate = _.template($('#status-messages-template').html());
+
+
+
+  tripTemplate = _.template($('#trip-template').html());
+
+  console.log(`About to fetch data from ${ tripList.url }`);
+
+// Register our update listener first, to avoid the race condition
+  tripList.on('update', renderTrips);
+  tripList.on('sort', renderTrips);
+
+// Fetch is what gets the data
+// When fetch gets back from the API call, it will add trips
+// to the list and then trigger an 'update' event
+  tripList.fetch();
+
+  console.log(tripList);
+
+// Sort Trip List by Trip Fields
+// Add a click handler for each of the table headers
+//to sort the table by that column
+
+  DISPLAY_TRIP_FIELDS.forEach((field) => {
+    const headerElement = $(`th.sort.${ field }`);
+    headerElement.on('click', (event) => {
```

Comment:
You also need to add something to style the selected column.  Right now it sorts, but doesn't indicate which column was sorted.  Adding a class which changes the styling of the `th` element would work well.